avm/res/resources/deployment-script/main.bicep

```diff
@@ -0,0 +1,256 @@
+metadata name = 'Deployment Scripts'
+metadata description = 'This module deploys Deployment Scripts.'
+metadata owner = 'Azure/module-maintainers'
+
+// ================ //
+// Definitions      //
+// ================ //
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type roleAssignmentType = {
+  @description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead.')
+  roleDefinitionIdOrName: string
+
+  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
+  principalId: string
+
+  @description('Optional. The principal type of the assigned principal ID.')
+  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device' | null)?
+
+  @description('Optional. The description of the role assignment.')
+  description: string?
+
+  @description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container"')
+  condition: string?
+
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
+}[]?
+
+// ================ //
+// Parameters       //
+// ================ //
+@description('Required. Name of the Deployment Script.')
+@maxLength(24)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. Specifies the Kind of the Deployment Script.')
+@allowed([
+  'AzureCLI'
+  'AzurePowerShell'
+])
+param kind string
+
+@metadata({
+  example: '''
+  {
+    '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myID': {}
+    '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myID2': {}
+  }
+  '''
+})
+@description('Optional. Specifies the user assigned identities to be assigned to the Deployment Script.')
+param userAssignedIdentities object?
```

Comment:
As per the CARML module & AVM specs, please use the UDT for this property

Reply:
In CARML, this is already done. Please refer to the already updated module [here](https://github.com/Azure/ResourceModules/blob/main/modules/resources/deployment-script/main.bicep#L8-L9).